swap/src/main.rs

```diff
@@ -206,13 +207,12 @@ async fn main() -> Result<()> {
 
 fn init_settings() -> Result<Settings> {
     let config_path = default_config_path()?;
-    let config = match read_config(config_path.clone()) {
+    let config = match read_config(config_path.clone())? {
         Ok(config) => config,
-        Err(config::Error::ConfigNotInitialized) => {
+        Err(ConfigNotInitialized {}) => {
             initial_setup(config_path.clone(), query_user_for_initial_testnet_config)?;
-            read_config(config_path)?
+            read_config(config_path)?.expect("after initial setup config can be read")
```

Comment:
`init_settings` returns a result. Return a result instead of using `expect`.

Reply:
It returns a nested result, where the outer is an actual error we want to bubble up, and the inner is `ConfigNotInitialized`. After running `initial_setup` we expect the config to be present. The expect makes this explicit.